presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -7549,6 +7549,80 @@ public void testSelectFromViewWithoutDefaultCatalogAndSchema()
         assertQuery(sessionNoCatalog, "SELECT count(*) FROM hive.tpch." + viewName, "VALUES 1");
     }
 
+    @Test(invocationCount = 100)
+    public void testSelectFromPrestoViewReferencingHiveTableWithTimestamps() {
+        // Hive views tests covered in TestHiveViews.testTimestampHiveView and TestHiveViesLegacy.testTimestampHiveView
+        String hiveTableName = "select_from_presto_view_hive_table_timestamps_table_" + randomTableSuffix();
+        assertUpdate(
+                withTimestampPrecision(getSession(), HiveTimestampPrecision.NANOSECONDS),
+                "CREATE TABLE " + hiveTableName + " AS SELECT TIMESTAMP '1990-01-02 12:13:14.123456789' ts",
+                1);
+
+        Session defaultSession = getSession();
+        Session millisSession = Session.builder(getSession())
+                .setCatalogSessionProperty("hive", "timestamp_precision", "MILLISECONDS")
+                .setCatalogSessionProperty("hive_timestamp_nanos", "timestamp_precision", "MILLISECONDS")
+                .build();
+        Session nanosSessions = Session.builder(getSession())
+                .setCatalogSessionProperty("hive", "timestamp_precision", "NANOSECONDS")
+                .setCatalogSessionProperty("hive_timestamp_nanos", "timestamp_precision", "NANOSECONDS")
+                .build();
+
+        // Presto view created with config property set to MILLIS and session property not set
+        String prestoViewNameDefault = "select_from_presto_view_hive_table_timestamps_default_view_" + randomTableSuffix();
+        assertUpdate(defaultSession, "CREATE VIEW " + prestoViewNameDefault + " AS SELECT *  FROM " + hiveTableName);
+
+        assertQuery(defaultSession, "SELECT CAST(ts as VARCHAR) FROM " + prestoViewNameDefault, "VALUES ('1990-01-02 12:13:14.123')");
+        assertThatThrownBy(
+                // TODO Presto view schema is fixed on creation
+                () ->assertQuery(defaultSession, "SELECT CAST(ts as VARCHAR) FROM hive_timestamp_nanos.tpch." + prestoViewNameDefault, "VALUES ('1990-01-02 12:13:14.123456789')")
```

Comment:
nit: no space after arrow (did checkstyle catch this?)

Reply:
Yeah - checkstyle fails on that one